internal/integration/unified/context.go

```diff
@@ -50,9 +50,6 @@ func newTestContext(
 
 func addFailPoint(ctx context.Context, failPoint string, client *mongo.Client) error {
 	failPoints := ctx.Value(failPointsKey).(map[string]*mongo.Client)
-	if _, ok := failPoints[failPoint]; ok {
-		return fmt.Errorf("fail point %q already exists in tracked fail points map", failPoint)
-	}
```

Comment:
It could be the case that a test sets a failpoint and disables it in a later operation, e.g. min pool "[Network error on minPoolSize background creation](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/tests/unified/minPoolSize-error.yml#L24)". In any event, if multiple failpoints are set with the same name, they will overwrite, so this check is unecessary.